### ISSUE TITLE
Exhaustiveness check fails on types with a literal field

### ISSUE THREAD

**Issue by ekisu** [microsoft/TypeScript#54095]
# Bug Report

### 🔎 Search Terms

exhaustiveness check, type alias

### 🕗 Version & Regression Information

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about exhaustiveness checks

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?ts=5.1.0-dev.20230501&ssl=14&ssc=2&pln=1&pc=1#code/C4TwDgpgBAwglgJwMYBtoF4oG8BQUoDWcAdgCYBcUAREoqhFQNx5QICGpcArgM6XFcAtgCMICZgF8cOAGZdiSYHAD2xKAHMIwAIIIIbABS1kaSvBMQAlNhY8A7nGBIAFlCN00AOiJlrufPhIbDzQNB4M5CwBrFpcCGoAsmzAzp4ACgCSUABUUMb0nuycvDm5AEzM0aQQMmxcKMCR0YGqPMBQAPoQAB7OdW1wAG4QMM4QSAT8EMMIUJj5aJXNesBxal29-UrDo+MES1ISQA)

### 💻 Code

```ts
type Circle = {
  kind: "circle";
  radius: number;
}

function getArea(circle: Circle) {
  switch (circle.kind) {
    case "circle":
      return Math.PI * circle.radius ** 2;
    default:
      const _exhaustiveCheck: never = circle;
      return _exhaustiveCheck;
  }
}
```

### 🙁 Actual behavior

An error is reported on the `_exhaustiveCheck` assignment: `Type 'Circle' is not assignable to type 'never'.`

### 🙂 Expected behavior

No error to be reported, as all possible instances of `Circle` have `circle.kind === "circle"`, which is covered by the first switch case

**Comment by fatcerberus** [comment]
Duplicate of #18056.  A single type isn't considered to be a union so discriminated union narrowing doesn't work.  This code *does* work in general, as long as you have more than one possibility:

[Playground](https://www.typescriptlang.org/play?ts=5.1.0-dev.20230501#code/C4TwDgpgBAwglgJwMYBtoF4oG8BQUoDWcAdgCYBcUAREoqhFQNx5QICGpcArgM6XFcAtgCMICZgF8cOUJCgBlAI5c2CDNhZEylKj2WqGzfDzikI-IaPE4pOAGZdiSYHAD2xKAHMIwAIJq2AAoeAAs2SEp4ZDQoAB8FfTUASg1jAHc4YCQQqGCwyAA6LVIU3Hx8JDYeaBo6NCpyFnLWHy4EDwBZNmAQgoAFAEkoACooUPCIAvZOXhHRgCYjcsrq6j0VNQam8rVgNo9xwpMzOahF7bM7Ni4UYEbmivceYCgAfQgADzDeFwA3CBgIQgSAI-Ag-wQUEwhwgjHwAHp4VBiK40lBMgByHhQVwEbb4Xb7N6fb7POD-QHAghLKQSIA)

Workaround for cases where there really is only one kind: Union the type with `{ kind: never }`.
